### DISCUSSION TITLE
Linear iteration residual

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#22188] [Q&A Modules: Phase field] [answered]
Hello, I am wondering what the residual function of the linear iteration is at each nonlinear iteration. Is it dx*J+R, where dx is the solution update, J is the Jacobian, and R is the nonlinear residual? Also, when using Newton as the solver_type, usually only one linear iteration is needed, but sometimes there are more than one linear iterations. Is that an indication of a wrong Jacobian in the kernels?

**Comment by hugary1995** [comment, marked as answer]
For your first question, almost -- it is preconditioned.

For your second question,  if you use a direct solver, usually only one linear iteration is needed, unless the matrix isn't well conditioned. If you use an iterative solver, then you need more than one linear iteration in general.

In short, if you use Newton + a direct solver + a full preconditioning matrix, more than one linear iteration is indeed an indication of either
- inaccurate Jacobian
- a poorly conditioned system

**Reply by GiudGiud** [reply]
Note there s a Python utility in moose (jacobian_analyzer) you may use to check your jacobian if you suspect it has issues

**Reply by GiudGiud** [reply]
Also residual is scaled if you use manual or automatic scaling

**Reply by xueyang94** [reply]
Yes I have used the Jacobian debugger before, thanks for pointing it out!
